subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -346,6 +352,7 @@ static int nct_provision(void)
 	nct.tls_config.hostname = NRF_CLOUD_HOSTNAME;
 
 #if defined(CONFIG_NRF_CLOUD_PROVISION_CERTIFICATES)
+#if defined(CONFIG_BSD_LIBRARY)
```

Comment:
That's  a big `#if` clause; I recommend splitting this function into two, one for provisioning using `nrf_inbuilt_key` and one using `tls_credential` instead.